builtin/providers/aws/auth_helpers.go

```diff
@@ -35,29 +35,32 @@ func GetAccountInfo(iamconn *iam.IAM, stsconn *sts.STS, authProviderName string)
 
 		metadataClient := ec2metadata.New(sess)
 		info, err := metadataClient.IAMInfo()
-		if err != nil {
-			// This can be triggered when no IAM Role is assigned
-			// or AWS just happens to return invalid response
-			return "", "", fmt.Errorf("Failed getting EC2 IAM info: %s", err)
+		if err == nil {
+			return parseAccountInfoFromArn(info.InstanceProfileArn)
+		}
+		// We can end up here if there's an issue with the instance metadata service
+		// or if we're getting credentials from AdRoll's Hologram (in which case IAMInfo will
+		// error out). In any event, if we can't get account info here, we should try
+		// the other methods available.
+		// If we have creds from something that looks like an IAM instance profile, but
+		// we were unable to retrieve account info from the instance profile, it's probably
+		// a safe assumption that we're not an IAM user
```

Comment:
I don't think this is a safe assumption we can make. The metadata API may be unavailable or broken and we'd like to inform the user in such case rather than falling through.